src/js/components/RadioButtonGroup/RadioButtonGroup.js

```diff
@@ -27,6 +29,10 @@ const RadioButtonGroup = forwardRef(
     ref,
   ) => {
     const formContext = useContext(FormContext);
+    const theme = useContext(ThemeContext) || defaultProps.theme;
+    const radioButtonGap = theme.global.radioButtonGroup
+      ? theme.global.radioButtonGroup.gap
+      : gap;
```

Comment:
I think that instead of fetching only the `gap` style from the theme, the behavior here should be more generic and should fetch all the Box props from the theme.
So essentially this const will be removed, and see my changes on line 110.